floc.bs

```diff
@@ -1 +1,178 @@
+
+  The {{InterestCohort}} dictionary is used to contain the [=string representation of the interest cohort id=] and the [=string representation of the interest cohort id=].
+
+  <pre class="idl">
+  dictionary InterestCohort {
+    DOMString id;
+    DOMString version;
+  };
+  </pre>
+</section>
+
+<section>
+  <h2 id="the-api">The API</h2>
+
+  The interest cohort API lives under the {{Document}} interface since the access permission is tied to the document scope, and the API is only available if the document is in [=secure context=].
+
+  <pre class="idl">
+    partial interface Document {
+        Promise&lt;InterestCohort&gt; interestCohort();
+    };
+  </pre>
+
+  The <dfn for="Document" method>interestCohort()</dfn> method steps are:
+  1. Let |p| be [=a new promise=].
+  1. Run the following steps [=in parallel=]:
+    1. If any of the following is true:
+        - [=this=] is not [=allowed to use=] the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" feature.
+        - The document is not allowed to access the interest cohort per user preference settings.
+        - The user agent believes that too many high-entropy bits of information have already been consumed by the given document, and exposing an interest cohort would violate a privacy budget.
+        - The <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a> is unavailable.
+
+        then:
+          1. [=Queue a task=] on the <dfn>interest cohort task source</dfn> to [=reject=] |p| with a "{{NotAllowedError}}" {{DOMException}}.
+          1. Abort these steps.
+    1. Let |id| be [=interest cohort id=] from running the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>.
+    1. Let |version| be the [=interest cohort version=] corresponding to the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>.
+    1. [=Queue a task=] on the [=interest cohort task source=] to perform the following steps:
+        1. Let |d| be the {{InterestCohort}} dictionary, with {{InterestCohort/id}} being the <a href="#string-representation-of-the-interest-cohort-id">string representation</a> of |id|, and {{InterestCohort/version}} being <a href="#string-representation-of-the-interest-cohort-version">string representation</a> of |version|.
+        1. [=Resolve=] |p| with |d|.
+  1. Return |p|.
+
+</section>
+
+<section>
+  <h2 id="interpretation-from-ad-techs">Interpretation from ad techs</h2>
+  Ad tech companies are supposed to observe the habits of each [=interest cohort=] and ad targeting can then be partly based on what group the person falls into. The browser vendors could publicly share more information about the [=interest cohort id=] (e.g. the range of numbers, whether they have semantics, etc.) or the [=interest cohort version=] (e.g. the algorithm detail, the compatibility between versions, etc.) to help the ad techs with their modeling decisions.
```

Comment:
Suggest s/Ad tech companies are supposed to observe/Organizations that wish to interpret cohorts can observe/

Reply:
While the API is intended for ad tech companies, the reality is that it's a public API and may be used by others. So let's replace ad tech with a more neutral term. 

e.g., s/ad techs with their modeling decisions/help with their modeling decisions/

Reply:
Done.